libpldmresponder/fru.cpp

```diff
@@ -915,40 +928,61 @@ void FruImpl::processFruPresenceChange(const DbusChangedProps& chProperties,
     }
 
     std::vector<std::string> portObjects;
-    static constexpr auto portInterface =
-        "xyz.openbmc_project.Inventory.Item.Connector";
 
-#ifdef OEM_IBM
-    if (fruInterface == "xyz.openbmc_project.Inventory.Item.PCIeDevice")
+    if (newPropVal)
     {
-        if (!pldm::responder::utils::checkFruPresence(fruObjPath.c_str()))
+        info("Presence changed to true for {PATH}", "PATH", fruObjPath);
+        if (fruInterface == "xyz.openbmc_project.Inventory.Item.PCIeDevice")
         {
-            return;
+#ifdef OEM_IBM
+            if (pldm::responder::utils::checkIfIBMCableCard(fruObjPath))
+            {
+                static constexpr auto portInterface =
+                    "xyz.openbmc_project.Inventory.Item.Connector";
+
+                // if the added fru is IBM cable card, build the adapter
+                // PDRs and the port PDRs.
+                buildIndividualFRU(fruInterface, fruObjPath);
+                pldm::responder::utils::findPortObjects(fruObjPath,
+                                                        portObjects);
+                info("Ports under the cable card: {NUM}", "NUM",
+                     (int)portObjects.size());
+                for (auto portObject : portObjects)
+                {
+                    buildIndividualFRU(portInterface, portObject);
+                }
+                return;
+            }
+            else
+            {
+                // PDRs realted to industry standard card are not
+                // rebuilt.
+                info("Skipping building fru for {PATH}", "PATH", fruObjPath);
```

Comment:
wouldn't these flood the journal ?

Reply:
Trace will be printed during slot disable operation triggered after poweron. I would like to keep this trace for few days for debug. We can remove it after a month or two